### ISSUE TITLE
Error: Failed to add alias

### ISSUE THREAD

**Issue by butlerjraines** [algorandfoundation/algokit-cli#585]
### Subject of the issue

Was trying to create a command line wallet alias and received the following error: Error: **Failed to add alias**

Example: algokit task wallet add -a G2ILI3MGMPCFOCIMOTHZSFMZSSXLYIPPB5PVQA3MVNPXTCPHDYDNXSQ75M howdy

This seems to be specific to my windows WSL Ubuntu local environment.

I have tried this on two separate machines.

If I install Algokit through Windows / Powershell the alias is created correctly with the same command

### Your environment

timestamp: 2024-10-20T19:09:39+00:00
AlgoKit: 2.4.2
AlgoKit Python: 3.12.3 (main, Sep 11 2024, 14:17:37) [GCC 13.2.0] (location: /home/username/.local/share/pipx/venvs/algokit)
OS: Linux-5.15.153.1-microsoft-standard-WSL2-x86_64-with-glibc2.39
docker: 27.2.0
docker compose: 2.29.2-desktop.2
git: 2.43.0
python: 3.12.3 (location: /usr/bin/python)
python3: 3.12.3 (location: /usr/bin/python3)
pipx: 1.4.3
poetry: 1.8.4
node: 22.9.0
npm: 10.8.3

### Steps to reproduce

1. In a similar environment 
2. Run this command algokit task wallet add -a G2ILI3MGMPCFOCIMOTHZSFMZSSXLYIPPB5PVQA3MVNPXTCPHDYDNXSQ75M howdy

### Expected behaviour

Alias 'howdy' added successfully.

### Actual behaviour

Error: Failed to add alias

**Comment by butlerjraines** [comment]
More information here. 

This seems to be an issue with keyrings. I've tried several troubleshooting steps:

raise NoKeyringError(msg)
keyring.errors.NoKeyringError: No recommended backend was available. Install a recommended 3rd party backend package; or, install the keyrings.alt package if you want to use the non-recommended backends. See https://pypi.org/project/keyring for details.
2024-10-21T06:55:05.177 algokit.core.tasks.wallet DEBUG Failed to add alias to keyring

After trying the keyring.alt package:

ModuleNotFoundError: No module named 'keyrings'

**Comment by neilcampbell** [comment]
Yeah it appears the WSL2 environment can't find an available keyring backend.
Have you tried this https://github.com/jaraco/keyring/issues/566#issuecomment-1792544475?
Let us know how you go and what you did so we can add to the docs.

**Comment by neilcampbell** [comment]
@butlerjraines Did you get a chance to try the above approach? I'm curious if it worked?

**Comment by neilcampbell** [comment]
kmd is the mechanism that come with algod. You can use kmd directly, as it's available in LocalNet. The trouble is that LocalNet is intentionally treated as a somewhat ephemeral resettable network, so on reset you loose all state. As a result we decided to use keyring which persists across LocalNet resets.

**Comment by butlerjraines** [comment]
So sorry for the delay. Thank for you patience.

This worked perfectly! Hopefully this helps someone else out.

**Comment by neilcampbell** [comment]
No worries at all, thanks for looping back in.
I'll add something to the docs to make it more discoverable.
